pandas/core/sorting.py

```diff
@@ -218,7 +218,7 @@ def decons_group_index(comp_labels, shape):
     x = comp_labels
     for i in reversed(range(len(shape))):
         labels = (x - y) % (factor * shape[i]) // factor
-        np.putmask(labels, comp_labels < 0, -1)
```

Comment:
np.putmask is significantly more performatn

Reply:
isn't this what you'd requested in https://github.com/pandas-dev/pandas/pull/37586#discussion_r519237490 ? Have reverted anyway